x-pack/plugins/secops/public/components/draggables/index.tsx

```diff
@@ -17,20 +17,77 @@ import { Provider } from '../timeline/data_providers/provider';
 export interface DefaultDraggableType {
   id: string;
   field: string;
-  value: string | null | undefined;
+  value?: string | null;
   name?: string | null;
   queryValue?: string | null;
   children?: React.ReactNode;
+  tooltipContent?: React.ReactNode;
 }
 
+/**
+ * Only returns true if the specified tooltipContent is exactly `null`.
+ * Example input / output:
+ * `bob -> false`
+ * `undefined -> false`
+ * `<span>thing</span> -> false`
+ * `null -> true`
+ */
+export const tooltipContentIsExplicitlyNull = (tooltipContent?: React.ReactNode): boolean =>
+  tooltipContent === null; // an explicit / exact null check
+
+/**
+ * Derives the tooltip content from the field name if no tooltip was specified
+ */
+export const getDefaultWhenTooltipIsUnspecified = ({
+  field,
+  tooltipContent,
+}: {
+  field: string;
+  tooltipContent?: React.ReactNode;
+}): React.ReactNode => (tooltipContent != null ? tooltipContent : field);
+
+/**
+ * Renders the content of the draggable, wrapped in a tooltip
+ */
+const Content = pure<{
+  children?: React.ReactNode;
+  field: string;
+  tooltipContent?: React.ReactNode;
+  value?: string | null;
+}>(({ children, field, tooltipContent, value }) =>
+  !tooltipContentIsExplicitlyNull(tooltipContent) ? (
+    <EuiToolTip
+      data-test-subj={`${field}-tooltip`}
+      content={getDefaultWhenTooltipIsUnspecified({ tooltipContent, field })}
+    >
+      <>{children ? children : value}</>
+    </EuiToolTip>
+  ) : (
+    <>{children ? children : value}</>
+  )
+);
+
+/**
+ * Draggable text (or an arbitrary visualization specified by `children`)
+ * that's only displayed when the specified value is non-`null`.
+ *
+ * @param id - a unique draggable id, which typically follows the format `${contextId}-${eventId}-${field}-${value}`
+ * @param field - the name of the field, e.g. `network.transport`
+ * @param value - value of the field e.g. `tcp`
+ * @param name - defaulting to `field`, this optional human readable name is used by the `DataProvider` that represents the data
+ * @param children - defaults to displaying `value`, this allows an arbitrary visualization to be displayed in lieu of the default behavior
+ * @param tooltipContent - defaults to displaying `field`, pass `null` to
+ * prevent a tooltip from being displayed, or pass arbitrary content
+ * @param queryValue - defaults to `value`, this query overrides the `queryMatch.value` used by the `DataProvider` that represents the data
+ */
 export const DefaultDraggable = pure<DefaultDraggableType>(
-  ({ id, field, value, name, children, queryValue }) =>
+  ({ id, field, value, name, children, tooltipContent, queryValue }) =>
     value != null ? (
       <DraggableWrapper
         dataProvider={{
           and: [],
           enabled: true,
-          id: escapeDataProviderId(`${id}-${field}`),
+          id: escapeDataProviderId(id),
```

Comment:
Did you want this to be less specific of an id? What if I use two `DefaultDragables` within the sam row renderer which will have the same `id`?

Previously, that was the thinking during my pull request to ensure that it was unique enough to have two draggables within the same row renderer.

Reply:
Oh I see, because below we have:

```
id={`${contextId}-${eventId}-${field}-${value}`}
```

it should be unique enough.